app/scripts/lib/util.ts

```diff
@@ -480,3 +481,97 @@ export function getConversionRatesForNativeAsset({
 
   return conversionRateResult;
 }
+
+// Cache for known domains
+let knownDomainsSet: Set<string> | null = null;
+let initPromise: Promise<void> | null = null;
+
+/**
+ * Initialize the set of known domains from the chains list
+ */
+export async function initializeRpcProviderDomains(): Promise<void> {
+  if (initPromise) {
+    return initPromise;
+  }
+
+  initPromise = (async () => {
+    try {
+      const chainsList = await getSafeChainsList();
+      knownDomainsSet = new Set<string>();
+
+      for (const chain of chainsList) {
+        if (chain.rpc && Array.isArray(chain.rpc)) {
+          for (const rpcUrl of chain.rpc) {
+            try {
+              // Removed cleaning step - will fail for URLs like:
+              // https://${INFURA_API_KEY}.infura.io/v3/
+              const url = new URL(rpcUrl);
+              knownDomainsSet.add(url.hostname.toLowerCase());
+            } catch (e) {
+              // Skip invalid URLs
+              continue;
+            }
+          }
+        }
+      }
+    } catch (error) {
+      console.error('Error initializing known domains:', error);
+      knownDomainsSet = new Set<string>();
+    }
+  })();
+
+  return initPromise;
+}
+
+/**
+ * Check if a domain is in the known domains list
+ *
+ * @param domain - The domain to check
+ */
+export function isKnownDomain(domain: string): boolean {
+  return knownDomainsSet?.has(domain?.toLowerCase()) ?? false;
```

Comment:
`domain` should always be defined here, per the type definition.

```suggestion
  return knownDomainsSet?.has(domain?.toLowerCase()) ?? false;
```